src/Nancy/Routing/Trie/RouteResolverTrie.cs

```diff
@@ -62,19 +62,15 @@ public void BuildTrie(IRouteCache cache)
         /// <returns>An array of <see cref="MatchResult"/> elements</returns>
         public MatchResult[] GetMatches(string method, string path, NancyContext context)
         {
-            if (string.IsNullOrEmpty(path))

```

Comment:
I think @jchannon is correct. Since you remove the null-check on `path` it can explode below on line 72 ?